app_dart/lib/src/service/firestore.dart

```diff
@@ -67,24 +72,88 @@ class FirestoreService {
     return databasesDocumentsResource.commit(commitRequest, kDatabase);
   }
 
-  Future<List<firestore.Task>> queryCommitTasks(String commitSha) async {
+  /// Returns all tasks running against the speificed [commitSha].
+  Future<List<Task>> queryCommitTasks(String commitSha) async {
+    final Map<String, Object> filterMap = <String, Object>{
+      '$kTaskCommitShaField = ': commitSha,
+    };
+    final List<Document> documents = await query(kTaskCollectionId, filterMap);
+    return documents.map((Document document) => Task.fromDocument(taskDocument: document)).toList();
+  }
+
+  /// Returns Firestore [Value] based on corresponding object type.
+  Value getValueFromFilter(Object comparisonOject) {
+    if (comparisonOject is int) {
+      return Value(integerValue: comparisonOject.toString());
+    } else if (comparisonOject is bool) {
+      return Value(booleanValue: comparisonOject);
+    }
+    return Value(stringValue: comparisonOject as String);
+  }
+
+  /// Generates Firestore query filter based on "human" read conditions.
+  Filter generateFilter(Map<String, Object> filterMap, String compositeFilterOp) {
+    final List<Filter> filters = <Filter>[];
+    filterMap.forEach((filterString, comparisonOject) {
+      final List<String> parts = filterString.split(' ');
+      if (parts.length != 2 || !kRelationMapping.containsKey(parts[1])) {
+        throw ArgumentError("Invalid filter string '$filterString'.");
+      }
+      final String name = parts[0];
+      final String comparison = kRelationMapping[parts[1]]!;
+      final Value value = getValueFromFilter(comparisonOject);
+
+      filters.add(
+        Filter(
+          fieldFilter: FieldFilter(
+            field: FieldReference(fieldPath: name),
+            op: comparison,
+            value: value,
+          ),
+        ),
+      );
+    });
+    Filter? finalFilter;
+    if (filters.length == 1) {
+      finalFilter = filters.single;
+    } else {
+      finalFilter = Filter(
+        compositeFilter: CompositeFilter(
+          filters: filters,
+          op: compositeFilterOp,
+        ),
+      );
+    }
+    return finalFilter;
+  }
+
+  /// Wrapper to simplify Firestore query.
+  ///
+  /// The [filterMap] follows format:
+  ///   {
+  ///     'fieldInt =': 1,
```

Comment:
Is this convention from firestore?

Reply:
No. It's a convention from datastore. This is actually a user side behavior. Using the same as before to make it same for user experience.
https://github.com/flutter/cocoon/blob/main/app_dart/lib/src/service/datastore.dart#L215
https://github.com/dart-lang/gcloud/blob/master/lib/src/db/db.dart#L185

Reply:
Thanks!